packages/actor-query-operation-group/lib/GroupsState.ts

```diff
@@ -86,11 +87,13 @@ export class GroupsState {
         await Promise.all(this.pattern.aggregates.map(async(aggregate) => {
           // If distinct, check first whether we have inserted these values already
           if (aggregate.distinct) {
-            const hash = this.hashBindings(bindings);
-            if (this.distinctHashes!.get(groupHash)!.has(hash)) {
+            const bindingsHash = this.hashBindings(bindings);
+            const hashBindingsAndAggregate = createHash('sha256')
+              .update(aggregate.variable.value).update(bindingsHash).digest().toString();
```

Comment:
Instead of including the aggregate variable name, wouldn't it also be possible to just scope the hashes to each variable separately. E.g. `distinctHashes: Record<variable name, Map<BindingsHash, Set<BindingsHash>>>`